### PR TITLE
Scaled out instance shows as one with tooltips for all the Hosts

### PR DESCRIPTION
**Pull request by SeanFeldman**
This addresses issue [Endpoint explorer in SI hard to use when repeated endpoint name appear](https://github.com/Particular/FeatureDevelopment/issues/310)
Should `EndpointExplorerTests` be updated to reflect this?

@Particular/ServiceInsight please review

### CODE REVIEW

**Comment by petersgiles** [comment]
If we have to make a lot of distinct lists is might be worth including an extension method that uses a HashSet (very fast)

    public static IEnumerable<TSource> DistinctBy<TSource, TKey>(
        this IEnumerable<TSource> source,
        Func<TSource, TKey> keySelector)
    {
        var knownKeys = new HashSet<TKey>();
        return source.Where(element => knownKeys.Add(keySelector(element)));
    }

you could call it like this 

    var distinctList = sourceList.DistinctBy(x => x);

or in this case 

    var hostNames = instances.DistinctBy(instance => instance.HostDisplayName);
    var tooltip = String.Join(",", hostNames.ToArray());

File: src/ServiceInsight/Explorer/EndpointExplorer/EndpointExplorerViewModel.cs [hunk 1]
```diff
@@ -170,9 +170,14 @@ public void RefreshData()
...
             ServiceControlRoot.Children.Clear();
-            foreach (var endpoint in endpoints.OrderBy(e => e.Name))
+
+            var endpointInstancesGroupedByName = endpoints.GroupBy(x => x.Name);
+            foreach (var scaledOutEndpoint in endpointInstancesGroupedByName)
             {
-                ServiceControlRoot.Children.Add(new AuditEndpointExplorerItem(endpoint));
+                var instances = scaledOutEndpoint.ToList();
+                var hostNames = instances.Select(instance => instance.HostDisplayName);
+                var tooltip = string.Join(", ", hostNames);
```
Comment (johnsimons) [inline src/ServiceInsight/Explorer/EndpointExplorer/EndpointExplorerViewModel.cs]: Just wondering if it is better to stack them ? instead of comma delimited

File: src/ServiceInsight/SequenceDiagram/SequenceDiagramView.xaml [hunk 2]
```diff
@@ -114,7 +114,7 @@
                                         <Label Grid.Row="0" Grid.Column="1" Content="{Binding FullName}" Style="{StaticResource TooltipLabelStyle}" />
                                         <Label Grid.Row="1" Grid.Column="0" Content="NSB Version" Style="{StaticResource TooltipLabelStyle}" />
                                         <Label Grid.Row="1" Grid.Column="1" Content="{Binding Version}" Style="{StaticResource TooltipLabelStyle}" />
-                                        <Label Grid.Row="2" Grid.Column="0" Content="Host" Style="{StaticResource TooltipLabelStyle}" />
+                                        <Label Grid.Row="2" Grid.Column="0" Content="HostNames" Style="{StaticResource TooltipLabelStyle}" />
```
Comment (johnsimons) [inline src/ServiceInsight/SequenceDiagram/SequenceDiagramView.xaml]: Why this change? wouldn't in this diagram we would want to see the exact host that handled the message ?

File: src/ServiceInsight/Explorer/EndpointExplorer/EndpointExplorerViewModel.cs (hunk 1 above)
Comment (SeanFeldman) [inline src/ServiceInsight/Explorer/EndpointExplorer/EndpointExplorerViewModel.cs]: Have no preference and it wasn't specified how to separate between host names. Is stacking them would be better @johnsimons ?

File: src/ServiceInsight/SequenceDiagram/SequenceDiagramView.xaml (hunk 2 above)
Comment (SeanFeldman) [inline src/ServiceInsight/SequenceDiagram/SequenceDiagramView.xaml]: Probably crippled in with R# rename. Good catch @johnsimons

**Comment by johnsimons** [comment]
Thanks @SeanFeldman, when u have time can u fix the description so it looks pretty for the release notes
